context/amf_ran.go

```diff
@@ -73,8 +73,10 @@ func (ran *AmfRan) Remove() {
 			NfStatus: mi.NfStatusDisconnected, NfName: ran.GnbId,
 		},
 	}
-	if err := metrics.StatWriter.PublishNfStatusEvent(gnbStatus); err != nil {
-		ran.Log.Errorf("Could not publish NfStatusEvent: %v", err)
+	if metrics.Status() {
+		if err := metrics.StatWriter.PublishNfStatusEvent(gnbStatus); err != nil {
+			ran.Log.Errorf("Could not publish NfStatusEvent: %v", err)
+		}
```

Comment:
I see you are adding `if` statements (`if metrics.Status() {`) in the different places where `metrics.StatWriter.PublishNfStatusEvent(gnbStatus)` is called. What about adding a single `if` statement inside `PublishNfStatusEvent(msgEvent mi.MetricEvent)` such that there are changes in fewer number of files and looks cleaner?

Something like this in `metrics/kafka.go`:
```go
func (writer Writer) PublishNfStatusEvent(msgEvent mi.MetricEvent) error {
	if !metrics.Status() {
		logger.KafkaLog.Info("Kafka is disabled")
		return nil
	}
	if msg, err := json.Marshal(msgEvent); err != nil {
		return err
	} else {
		logger.KafkaLog.Debugf("publishing nf status event[%s] ", msg)
		if err := StatWriter.SendMessage(msg); err != nil {
			logger.KafkaLog.Errorf("Error publishing nf status event: %v", err)
		}
	}
	return nil
}
```

Reply:
I thought about this but it also means we are initializing Kafka code (Since to come here we need writer). I wanted to avoid that.

Reply:
I can do the check in metric in case of - PublishNfStatusEvent but similar check is not good for PublishUeCtxtEvent

Reply:
Got it.